llama-dev/llama_dev/test/__init__.py

```diff
@@ -157,17 +157,29 @@ def test(
         for r in results
         if r["status"] == ResultStatus.INSTALL_FAILED
     ]
-    skipped = [
+    skipped_no_tests = [
+        r["package"].relative_to(repo_root)
+        for r in results
+        if r["status"] == ResultStatus.SKIPPED and "package has no tests" in r["stderr"]
```

Comment:
While this works, the root problem is that `ResultStatus.SKIPPED` is returned both when `if not package_has_tests(package_path):` and when `if not is_python_version_compatible(package_data):` at lines 284 and 294. 

If we replace `ResultStatus.SKIPPED` with two new statuses like `NO_TESTS` and `UNSUPPORTED_PYTHON_VER` (or something like this), this check would be more robust.